src/ypd/servlets/user_servlet.py

```diff
@@ -1,27 +1,93 @@
+
+                    '''Recommended to create environment variables for mail_username and mail_password for security/convenience reasons'''                    
+                    current_app.config.update({"MAIL_SERVER": 'smtp.gmail.com', 'MAIL_PORT': 587, 'MAIL_USERNAME': os.environ.get('MAIL_USERNAME'), 'MAIL_PASSWORD': os.environ.get('MAIL_PASSWORD'), 'MAIL_DEFAULT_SENDER': os.environ.get('MAIL_USERNAME'), 'MAIL_USE_TLS' : True, 'MAIL_USE_SSL': False})
+                    mail = Mail()
+                    mail.init_app(current_app)
+                    mail.send_message(subject="YOUR PASSWORD HAS BEEN CHANGED!", recipients=[current_user.email], body=Markup(f"""<h2>Hello <b> {current_user.username} </b>, </h2> <br> Your password has been changed. <br> If this is not correct, please contact support!"""))
+
+                    flash("Please login again")
+                    return redirect(url_for('UserView:logout'))
+            except Exception as e:
+                flash(str(e))
+        return render_template('change_pwd.html', form=form)
+    
+    @route('/forgot_pwd/', methods=['POST', 'GET'])
+    def forgotPassword(self):
+        form = ValidateUserForm()
+        if form.validate_on_submit:
+            try:
+                user = User.get_by_username(form.username.data, form.email.data)
+                if user:
+                    login_user(user)
+                    return redirect(url_for('UserView:changePassword'))
+            except TypeError as e:
+                flash(str(e))
+        return render_template('forgot_password.html', form=form)
+    
+    # def forgotEmailOrUsername(self):
+    #     form = ValidateUsernameForm()
+    #     if form.validate_on_submit:
+    #         try:
+    #             user = User.get_by_username(username=form.username.data)
+    #             login_user(user)
+    #             return redirect(url_for('UserView:changePassword'))
+    #         except TypeError as e:
+    #             flash(str(e))
+    #     return render_template('forgot_password.html', form=form)
+
+    @login_required
+    def deleteAccount(self):
+        current_app.config.update({"MAIL_SERVER": 'smtp.gmail.com', 'MAIL_PORT': 587, 'MAIL_USERNAME': os.environ.get('MAIL_USERNAME'), 'MAIL_PASSWORD': os.environ.get('MAIL_PASSWORD'), 'MAIL_DEFAULT_SENDER': os.environ.get('MAIL_USERNAME'), 'MAIL_USE_TLS' : True, 'MAIL_USE_SSL': False})
+        mail = Mail()
+        mail.init_app(current_app)
+
+        '''Timed account deletion is still under construction'''
+        mail.send_message(subject="YOUR ACCOUNT IS NO LONGER ACTIVE!", recipients=[current_user.email], body=f"""<h2>Hello <b> {current_user.username} </b>, </h2> <br> Your account is no longer active and will be deleted within 3 days. <br> If this is not correct, please contact support!""")
+        current_user.delete_account()
+        return redirect(url_for('UserView:logout'))
+
```

Comment:
So the user can just delete their own account with no confirmation?
Usually the point of emails is so you can have two-factor authentication before the user performs an actions such as deleting their account or changing their password

Reply:
Fixed